utils/url.tsx

```diff
@@ -3,23 +3,33 @@
 
 import {latinise} from 'utils/latinise.jsx';
 
-export function cleanUpUrlable(input) {
-    var cleaned = latinise(input);
+type WindowObject = {
+    location: {
```

Comment:
You can use the provided `Location` type here.
https://developer.mozilla.org/en-US/docs/Web/API/Location

Reply:
Thanks @devinbinnie, I didn't know of that type.

But changing the type to that would require every caller to pass and object which fully satisfies the Location type, even though we are only interested in certain properties of it. For instance, the tests in `utils/url.test.tsx` would have to be heavily modified to implement that interface, but we wouldn't use any additional properties of it.

An alternate approach could be to make `WindowObject.location` a `Partial<Location>`, but then our logic would have to be updated to check every single property for its existence.
Of course we could use mapped types to create a Location type where all properties except the ones we are interested in are optional, but that would some hard to read code along the lines of:

```ts
type OptionalExceptFor<T, TRequired extends keyof T> = Partial<T> & Pick<T, TRequired>

type CustomLocation = OptionalExceptFor<Location, 'origin' | 'protocol' | 'hostname' | 'port'>;
```
(code adapted from [this](https://stackoverflow.com/questions/52703321/make-some-properties-optional-in-a-typescript-type) SO thread)

If constructs like that `OptionalExceptFor` are already in use in the project, then I would of course do it that way, but I haven't found one yet.

What are your thoughts on that?

Reply:
That seems like a lot of effort, let's just go with your defined type for now.

Reply:
Great, thanks.